table/src/main/java/com/bazaarvoice/emodb/table/db/astyanax/TableJson.java

```diff
@@ -335,6 +357,17 @@ Delta newNextState(String storageUuid, StorageState state, Object markerValue) {
                 .build();
     }
 
+    // Delta for storing the Table metadata change updates.
+    Delta newTableMetadataUpdate(String tableName, String updateType, String dateTime) {
+        Delta updateTypeDelta = Deltas.mapBuilder()
+                .put(dateTime, updateType)
+                .build();
+
+        return Deltas.mapBuilder()
+                .update(tableName, updateTypeDelta)
+                .build();
+    }
```

Comment:
Just a thought:  This produces documents such as:

```
  {
    "review:testcustomer": {
      "2018-03-23T20:55:34.551": "DROP"
    },
    "story:testcustomer": {
      "2018-03-23T20:56:18.539": "DROP"
    },
    "~id": "2018-03-23",
    "~table": "__system_sor:table_metadata_changes",
    ...
}
```

Maybe consider putting the content in a common top-level attribute.  My thinking is that it's cumbersome to separate what's data from what's intrinsics, and if they were under another attribute then it's clear that everything under it is table.  For example:

```
  {
    "tables": {
        "review:testcustomer": {
          "2018-03-23T20:55:34.551": "DROP"
        },
        "story:testcustomer": {
          "2018-03-23T20:56:18.539": "DROP"
        }
    },
    "~id": "2018-03-23",
    "~table": "__system_sor:table_metadata_changes",
    ...
}

```